lib/selinux/selinux_linux.go

```diff
@@ -131,6 +179,32 @@ func CheckConfiguration(ensureEnforced bool, logger *slog.Logger) error {
 	return nil
 }
 
+// this function attempts to diagnose why Teleport is not running under the correct SELinux domain.
+func diagnoseDomain(procCtx *selinuxContext, logger *slog.Logger) error {
+	if procCtx.user != seSystemUser || procCtx.role != seSystemRole {
+		logger.WarnContext(context.TODO(), "Teleport is not running as the system_u:system_r SELinux user and role, running Teleport as a Systemd service is recommended when --enable-selinux is passed", "selinux_context", procCtx.String())
+	}
+	fallbackErr := trace.Errorf("Teleport is running under the wrong SELinux domain %q instead of %q, SELinux will not enforce Teleport correctly", procCtx.domain, domain)
```

Comment:
What is the user supposed to do in this case? Can the error message give some steps?

Reply:
Added link to docs